dali/python/nvidia/dali/plugin/pytorch.py

```diff
@@ -107,45 +112,56 @@ def __next__(self):
             outputs.append(p._share_outputs())
         for i in range(self._num_gpus):
             dev_id = self._pipes[i].device_id
-            out_data = []
-            out_labels = []
-            # segregate outputs into data/label entries
+            # initialize dict for all output categories
+            category_outputs = {key : [] for key in self._output_categories}
+            # segregate outputs into categories
             for j, out in enumerate(outputs[i]):
-                if self.output_map[j] == "data":
-                    out_data.append(out)
-                elif self.output_map[j] == "label":
-                    out_labels.append(out)
+                category_outputs[self.output_map[j]].append(out)
 
             # Change DALI TensorLists into Tensors
-            data = [x.as_tensor() for x in out_data]
-            data_shape = [x.shape() for x in data]
+            category_tensors = dict()
+            category_shapes = dict()
+            for category, output_list in category_outputs.items():
+                category_tensors[category] = [x.as_tensor() for x in output_list]
+                category_shapes[category] = [x.shape() for x in category_tensors[category]]
+
             # Change label shape from [batch_size, 1] to [batch_size]
-            labels = [x.as_tensor() for x in out_labels]
-            for l in labels:
-                l.squeeze()
+            if "labels" in self._output_categories:
```

Comment:
I think that we should remove this hacky "squeeze only if the output is named `labels`" and leave it to the user to squeeze the output or not, outsite of the plugin.

Reply:
Sure, my comment targeted pytorch version only